patches/tModLoader/Terraria/Initializers/LaunchInitializer.cs.patch

```diff
@@ -22,6 +22,59 @@
  		}
  
  		private static void LoadClientParameters(Main game) {
+-			string iP;
+-			if ((iP = TryParameter("-j", "-join")) != null)
+-				game.AutoJoin(iP);
++			string ip;
++			if ((ip = TryParameter("-j", "-join")) != null) {
++
++				int port = -1;
++				if (ip.Contains(':')) {
++					int colonInd = ip.LastIndexOf(':');
++					if (int.TryParse(ip.Substring(colonInd + 1), out int portNum)) {
++						port = portNum;
++						ip = ip.Substring(0, colonInd);
++					}
++				}
++
++				if (!Netplay.SetRemoteIP(ip)) {
++					ModLoader.Logging.tML.Error("Could not set remote IP: " + ip);
++				}
++				else {
++					if (port >= 0)
++						Netplay.ListenPort = port;
++
++					ModLoader.ModLoader.OnSuccessfulLoad += () => {
++						string player = TryParameter("-plr", "-player");
++						IO.PlayerFileData fileData = null;
++						if (player is not null) {
++							if (System.IO.File.Exists(player)) {
++								fileData = Player.GetFileData(player, false);
++							}
++							else {
++								Main.LoadPlayers();
++								fileData = Main.PlayerList.FirstOrDefault(p => p.Name == player);
++							}
++						}
++
++						if (fileData is not null) {
++							fileData.SetAsActive();
++							Main.menuMode = 10;
++							Netplay.StartTcpClient();
++							return;
++						}
++
++						Main.OpenPlayerSelect(fileData => {
++							fileData.SetAsActive();
++							Main.menuMode = 10;
++							Netplay.StartTcpClient();
++						});
++					};
++				}
++			}
```

Comment:
Could this not all be in `Main.AutoJoin`?

Reply:
I can move code there. Should I add `string player` parameter to AutoJoin or make it parse -plr parameter?

Reply:
Add parameter